electron/api/runJourney.ts

```diff
@@ -0,0 +1,214 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+import path from 'path';
+import { writeFile, rm, mkdir } from 'fs/promises';
+import { BrowserWindow, IpcMainInvokeEvent } from 'electron';
+import logger from 'electron-log';
+import isDev from 'electron-is-dev';
+import type {
+  ActionInContext,
+  RecorderSteps,
+  RunJourneyOptions,
+  StepEndEvent,
+  StepStatus,
+  TestEvent,
+} from '../../common/types';
+
+import { JOURNEY_DIR, PLAYWRIGHT_BROWSERS_PATH } from '../config';
+import { SyntheticsManager } from '../syntheticsManager';
+
+export async function runJourney(
```

Comment:
This function needs some more attention. But I prefer to defer it as I need to add unit tests for this and that'd be better time doing that.